### ISSUE TITLE
Error: No component factory found for DialogComponent. Did you add it to @NgModule.entryComponents?

### ISSUE THREAD

**Issue by imchaitanya9** [ngneat/dialog#9]
## I'm submitting a...

<pre><code>
[x] Support request
[x] Bug report
</code></pre>

## Current behavior

The component which is used for dialogBox, which is already available in entryComponents and also consider to be added into LazyModules. In this case the injector is not able to find the component we want to show into a modal.

Error:
<pre><code>
ERROR
Error: No component factory found for DialogComponent. Did you add it to @NgModule.entryComponents?
</code></pre>
StackBlits Example link: https://stackblitz.com/edit/angular-8-app-example-cn5vja?file=src/duplicate/duplicate.component.ts

## Expected behavior

The dialogbox component should get close/unmount via `this.dialogRef.close()` without any error.

## Environment

<pre><code>
Angular version: ~8.2.14
 
For Tooling issues:
- Node version: 12.18.1
- Platform:  Mac

**Comment by NetanelBasal** [comment]
@tonivj5 we do not support versions 9-? We should add it to the entry components.

**Comment by imchaitanya9** [comment]
> @tonivj5 we do not support versions 9-? We should add it to the entry components.

we already have it into entry components

**Comment by NetanelBasal** [comment]
@imchaitanya9 as a quick workaround you can add it yourself to `AppModule`:

```ts
import { NgModule } from "@angular/core";
import { BrowserModule } from "@angular/platform-browser";

import { AppComponent } from "./app.component";
import { DuplicateComponent } from "../duplicate/duplicate.component";
import { DialogModule, DialogComponent } from "@ngneat/dialog";

@NgModule({
  imports: [BrowserModule, DialogModule.forRoot()],
  declarations: [AppComponent, DuplicateComponent],
  entryComponents: [DuplicateComponent, DialogComponent],
  bootstrap: [AppComponent]
})
export class AppModule {}
```

**Comment by imchaitanya9** [comment]
> @imchaitanya9 as a quick workaround you can add it yourself to `AppModule`:
> 
> ```ts
> import { NgModule } from "@angular/core";
> import { BrowserModule } from "@angular/platform-browser";
> 
> import { AppComponent } from "./app.component";
> import { DuplicateComponent } from "../duplicate/duplicate.component";
> import { DialogModule, DialogComponent } from "@ngneat/dialog";
> 
> @NgModule({
>   imports: [BrowserModule, DialogModule.forRoot()],
>   declarations: [AppComponent, DuplicateComponent],
>   entryComponents: [DuplicateComponent, DialogComponent],
>   bootstrap: [AppComponent]
> })
> export class AppModule {}
> ```

Thanks for workaround!

however, I got one more error:
```
ERROR
Error: StaticInjectorError(AppModule)[DuplicateComponent -> DialogRef]:
StaticInjectorError(Platform: core)[DuplicateComponent -> DialogRef]:
NullInjectorError: No provider for DialogRef!
```

**Comment by NetanelBasal** [comment]
You're using template and it doesn't have DialogRef.

**Comment by NetanelBasal** [comment]
You should do: `this.dialog.open(DuplicateComponent)`

**Comment by imchaitanya9** [comment]
> You should do: `this.dialog.open(DuplicateComponent)`

I also have to handle `input`, `output` for this component.

**Comment by brunano21** [comment]
@NetanelBasal @tonivj5 hey guys, any update on this?

**Comment by NetanelBasal** [comment]
There is no issue here. You should use a component if you want a reference to `DialogRef`. If you want to use it in the template you can do this:

```html
    <ng-template #modalTpl let-ref>
      <h1>Hello World</h1>

      <button (click)="ref.close()">Close</button>
    </ng-template>
```

**Comment by NetanelBasal** [comment]
Or did you mean adding it to the entry component?

**Comment by NetanelBasal** [comment]
Released.

**Comment by brunano21** [comment]
@NetanelBasal Hey, I have tried v1.0.3 but I think we still face the same problem. 

Let us recap what we notice because I think we created a bit of confusion, apologies for that. So I have created two stackblitz:
1) Works as expected: https://angular-8-app-example-q5erk3.stackblitz.io
2) Not working: https://angular-8-app-example-auftkp.stackblitz.io

Both apps have a lazy loaded module, a LazyComponent and a `DialogContent` component
In 1) the app module `DialogContent` component is added to the `app.module`'s `declarations` array and into the `entryComponents` array.
In 2) the `app.module` is _clean_, whilst the `lazy.module`'s `declarations` array contains the `DialogContent` and we added the same into the `entryComponents` array.

In the second stackblitz, when we click on "Show dialog" button, nothing happens and an error is thrown:
```
ERROR Error: No component factory found for DialogContent. Did you add it to @NgModule.entryComponents?
```
